pkg/messenger/messenger.go

```diff
@@ -0,0 +1,34 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package messenger
+
+// The messenger package is responsible for the handling of communication between agents.
+// It takes care of threading, timing, etc.
+// Each message that we are going to send should be map[string]interface{} type.
+// e.g we have type model.Ack{Type: "type", ID: "id"}
+// it should be converted to: map[@id:id @type:type]
+// NOTE: The package modifies message data by JSON tag name according to aries-rfcs.
+//       Fields like @id, ~thread , etc. are redundant and may be rewritten.
+
+// Client provides methods for the communication
+type Client interface {
+	// HandleMessage prepares data needed for the communication according to the options.
+	// Options should be provided by the service.
+	HandleMessage(msg map[string]interface{}, opts ...interface{})
+
+	// ReplyTo replies to the message by given msgID.
+	// Keeps threadID in the *decorator.Thread.
+	// Using this function means that communication will be on the same thread.
+	ReplyTo(msgID string, msg map[string]interface{}, myDID, theirDID string)
```

Comment:
Seems like the inclusion of `myDID` and `theirDID` are superfluous if a user has the `msgID`.

Reply:
Agree!

Reply:
also, I have changed `ReplyToNested(msgID string, msg map[string]interface{})`